### PR TITLE
Add PrettierESLintLinter

### PR DESCRIPTION
**Pull request by CrabDude**

### CODE REVIEW

**Comment by jparise** [comment]
Could we instead adjust the relative priorities of these two linters so they always run in a predictable order?

File: src/PrettierESLintLinter.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  public function getDefaultInterpreter() {
+    return __DIR__ . '/node4_proxy';
+  }
+
+  public function getDefaultBinary() {
+    echo $this->getProjectRoot() . '/' . $this->cwd;
+    list($err, $stdout, $stderr) = exec_manual('yarn -s --cwd %s which prettier-eslint', $this->getProjectRoot() . '/' . $this->cwd);
+    $binaryPath = strtok($stdout, "\n");
+    echo $binaryPath;
```
Comment (jparise) [inline src/PrettierESLintLinter.php]: Debugging `echo`

File: src/PrettierESLintLinter.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  public function getLinterConfigurationName() {
+    return 'prettier-eslint';
+  }
+
+  public function shouldUseInterpreter() {
+    return true;
+  }
+
+  public function getDefaultInterpreter() {
+    return __DIR__ . '/node4_proxy';
```
Comment (jparise) [inline src/PrettierESLintLinter.php]: Could this just be `node` with `--harmony_array_includes` as a mandatory flag?

I don't love shipping the helper script.

File: __phutil_library_map__.php [hunk 3]
```diff
@@ -11,8 +11,8 @@
   'class' => array(
     'ApacheThriftGeneratedLinter' => 'src/ApacheThriftGeneratedLinter.php',
     'ApacheThriftLinter' => 'src/ApacheThriftLinter.php',
-    'PrettierLinter' => 'src/PrettierLinter.php',
```
Comment (jparise) [inline __phutil_library_map__.php]: Removed `PrettierLinter` by mistake?

Comment (CrabDude) [reply __phutil_library_map__.php]: Fixed & sorted.

**Comment by CrabDude** [comment]
This is good to go. Includes 2 edge case fixes:
1. `node` not installed (now show install instructions instead of exception)
2. PHP@<5.4 (Array literals `[]` not supported)

Includes 2 improvements:
1. Minor README fix
2. Better logging when wrong/no version of yarn  installed

File: src/PrettierESLintLinter.php [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      return __DIR__ . '/node4_proxy';
+    }
+    // Also used if node doesn't exist, causing an install instructions error
+    return 'node';
+  }
+
+  public function getDefaultBinary() {
+    list($err, $stdout, $stderr) = exec_manual('yarn -s --cwd %s which prettier-eslint', $this->getProjectRoot() . '/' . $this->cwd);
+    $binaryPath = strtok($stdout, "\n");
+    if ($binaryPath == "") {
```
Comment (jparise) [inline src/PrettierESLintLinter.php]: `if (empty($binaryPath)) {` is a little more idiomatic.

File: src/PrettierESLintLinter.php [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+  protected function parseLinterOutput($path, $err, $stdout, $stderr) {
+    if ($err) {
+      return false;
+    }
+
+    $originalText = $this->getData($path);
+    $messages = array();
+
+    // Ignored files have empty $stdout
```
Comment (jparise) [inline src/PrettierESLintLinter.php]: "having" ?

File: src/PrettierESLintLinter.php (hunk 4 above)
Comment (CrabDude) [reply src/PrettierESLintLinter.php]: Fixed.

File: src/PrettierESLintLinter.php (hunk 5 above)
Comment (CrabDude) [reply src/PrettierESLintLinter.php]: Clarified.
> // Note: $stdout is empty for ignored files

**Pull request merged**